client/src/lib/components/sidebar/SideBar.svelte

```diff
@@ -15,6 +17,20 @@
 			css: (_t, u) => `margin-right: ${u * width * -1}px;`
 		};
 	}
+
+	/**
+	 * Returns the title for this page as defined by page path in lib/assets/other/Titles.json.
+	 * @return {string} Title of the page in Titles.json if defined, "YAPms" if title not defined.
+	 */
+	function getTitle(): string {
+		const obj = titles.find((elem) => elem.path === $page.url.pathname); //Find title object corresponding to path
+		if (obj?.title !== undefined) {
+			//Returns undefined if obj or obj.title undefined.
+			return obj.title;
+		} else {
+			return 'YAPms'; //If no title in Titles.json for this page, "YAPms" is the title!
+		}
+	}
```

Comment:
I remember, you saying you'd be worried about the performance of the linear search.

We could switch this function to a reactive statement instead.
```
$: title = titles.find((elem) => elem.path === $page.url.pathname)?.title ?? 'YAPms';
```

The $: will create a reactive statement. This statement will update whenever any variable inside it changes, in this case whenever $page.url.pathname changes. Meaning it might be a little bit more performant.

Also the code is a bit more succinct.

Let me know if you wanna use the reactive statement or not.

Reply:
This is awesome notation! Wish I knew about it before hand. I've changed to the reactive statement.